src/test/java/org/cactoos/scalar/XorTest.java

```diff
@@ -0,0 +1,169 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.scalar;
+
+import org.cactoos.Scalar;
+import org.cactoos.iterable.IterableOf;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.ScalarHasValue;
+
+/**
+ * Test case for {@link Xor}.
+ * @since 0.48
+ */
+@SuppressWarnings({"PMD.AvoidDuplicateLiterals", "PMD.TooManyMethods"})
+final class XorTest {
+
+    @Test
+    void trueTrue() throws Exception {
+        new Assertion<>(
+            "Either one, but not both nor none",
+            new Xor(
+                new True(),
+                new True()
+            ),
+            new ScalarHasValue<>(false)
+        ).affirm();
+    }
+
+    @Test
+    void falseTrue() throws Exception {
+        new Assertion<>(
+            "Either one, but not both nor none",
+            new Xor(
+                new False(),
+                new True()
+            ),
+            new ScalarHasValue<>(true)
+        ).affirm();
+    }
+
+    @Test
+    void treuFalse() throws Exception {
+        new Assertion<>(
+            "Either one, but not both nor none",
+            new Xor(
+                new True(),
+                new False()
+            ),
+            new ScalarHasValue<>(true)
+        ).affirm();
+    }
+
+    @Test
+    void falseFalse() throws Exception {
+        new Assertion<>(
+            "Either one, but not both nor none",
+            new Xor(
+                new False(),
+                new False()
+            ),
+            new ScalarHasValue<>(false)
+        ).affirm();
+    }
+
+    @Test
+    void singleTrue() throws Exception {
+        new Assertion<>(
+            "Single True must be True",
+            new Xor(
+                new True()
+            ),
+            new ScalarHasValue<>(true)
+        ).affirm();
+    }
+
+    @Test
+    void singleFalse() throws Exception {
+        new Assertion<>(
+            "Single False must be False",
+            new Xor(
+                new False()
+            ),
+            new ScalarHasValue<>(false)
+        ).affirm();
+    }
+
+    @Test
+    void emptyIterator() throws Exception {
+        new Assertion<>(
+            "Empty iterator must be true",
```

Comment:
@kokodyn iterable, not iterator :) same with the test name

Reply:
@victornoel fixed